device/common/include/traccc/finding/device/impl/find_tracks.ipp

```diff
@@ -271,41 +308,48 @@ TRACCC_HOST_DEVICE inline void find_tracks(
-                .step = payload.step,
-                .previous_candidate_idx = in_param_id,
-                .meas_idx = std::numeric_limits<unsigned int>::max(),
-                .seed_idx = in_param_id,
-                .n_skipped = 1,
-                .chi2 = std::numeric_limits<traccc::scalar>::max()};
-        } else {
-            const unsigned int prev_link_idx =
-                payload.prev_links_idx + in_param_id;
 
-            const candidate_link& prev_link = links.at(prev_link_idx);
+        const unsigned int prev_link_idx = payload.prev_links_idx + in_param_id;
+        const unsigned int seed_idx =
+            payload.step == 0 ? in_param_id : links.at(prev_link_idx).seed_idx;
+
+        vecmem::device_atomic_ref<unsigned int> num_tracks_per_seed(
+            n_tracks_per_seed.at(seed_idx));
+        const unsigned int s_pos = num_tracks_per_seed.fetch_add(1);
+
+        if (s_pos < cfg.max_num_branches_per_seed) {
+            const unsigned int n_skipped =
+                payload.step == 0 ? 0 : links.at(prev_link_idx).n_skipped;
+
+            if (n_skipped >= cfg.max_num_skipping_per_cand || last_step) {
+                const unsigned int n_cands = payload.step - n_skipped;
+                if (n_cands >= cfg.min_track_candidates_per_track) {
+                    // In case of max skipping and min length being 0, and first
+                    // step being skipped, the links are empty, and the tip has
+                    // nowhere to point
+                    assert(payload.step > 0);
+                    tips.push_back(prev_link_idx);
+                }
+            } else {
+                // Add measurement candidates to link
+                const unsigned int l_pos = links.bulk_append_implicit(1);
 
-            assert(payload.step == prev_link.step + 1);
+                links.at(l_pos) = {
+                    .step = payload.step,
+                    .previous_candidate_idx = prev_link_idx,
+                    .meas_idx = std::numeric_limits<unsigned int>::max(),
+                    .seed_idx = seed_idx,
+                    .n_skipped = n_skipped + 1,
+                    .chi2 = std::numeric_limits<traccc::scalar>::max()};
 
-            links.at(l_pos) = {
-                .step = payload.step,
-                .previous_candidate_idx = prev_link_idx,
-                .meas_idx = std::numeric_limits<unsigned int>::max(),
-                .seed_idx = prev_link.seed_idx,
-                .n_skipped = prev_link.n_skipped + 1,
-                .chi2 = std::numeric_limits<traccc::scalar>::max()};
+                out_params.at(l_pos - payload.curr_links_idx) =
+                    in_params.at(in_param_id);
+                out_params_liveness.at(l_pos - payload.curr_links_idx) = 1u;
+            }
         }
-
-        out_params.at(l_pos - payload.curr_links_idx) =
-            in_params.at(in_param_id);
-        out_params_liveness.at(l_pos - payload.curr_links_idx) = 1u;
     }
-}
```

Comment:
Found the problem! It was this bracket that caused all the namespace problems :laughing: